### ISSUE TITLE
Gradle plugin warning from GenerateClientStubsFromDslTask

### ISSUE THREAD

**Issue by rivancic** [spring-cloud/spring-cloud-contract#1773]
**Describe the bug**
One uses **org.springframework.cloud.contract** Gradle plugin.
Invoking **generateClientStubs** task with Gradle version 7.1 you will receive following warning
```
> Task :xxx:generateClientStubs
The JavaExecHandleBuilder.setMain(String) method has been deprecated. This is scheduled to be removed in Gradle 8.0. Please use the mainClass property instead. Consult the upgrading guide for further information: https://docs.gradle.org/7.4.2/userguide/upgrading_version_7.html#java_exec_properties
```

Source of the issue (spring-cloud-contract-gradle-plugin): https://github.com/spring-cloud/spring-cloud-contract/blob/8faa11750797c6350218ae444630c73b85c55f27/spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/java/org/springframework/cloud/contract/verifier/plugin/GenerateClientStubsFromDslTask.java#L94

See deprecation warning since Gradle 7.1 for JavaExec task

- API Documentation: https://docs.gradle.org/7.1/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:main
- Release Notes, Deprecations: https://docs.gradle.org/7.1/userguide/upgrading_version_7.html#java_exec_properties
- Gradle Issue: https://github.com/gradle/gradle/issues/17048
- Gradle commit: https://github.com/gradle/gradle/commit/5dc1bf4d3fc76791fb720d15a6d5d105e77bce48

**Sample**
Basically any Gradle (using Gradle 7.1 and newer) project with stubs defined. Will create sample project if necessary.

**Comment by marcingrzejszczak** [comment]
fixed via https://github.com/spring-cloud/spring-cloud-contract/pull/1774
